app/controllers/languages_controller.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class LanguagesController < ApplicationController
+  # protect_from_forgery prepend: true, with: :exception, except: :create
+  # skip_before_action :verify_authenticity_token, only: %i[interactive_render from_json]
+  # before_action :log_csrf_error, only: %i[create]
+  # before_action :set_response, only: %i[show preference destroy]
+  # before_action :set_locale, only: %i[show]
+
+  before_action :set_response, only: %i[change]
+
+  def show
+    @locale = current_user.locale
+    @callback = show_params[:cb]
+    @response_id = show_params[:r_id]
+  end
+
+  def change
+    @locale = change_params[:locale]
+    unless current_user.update locale: @locale == 'nl' ? 'nl' : 'en'
+      Rails.logger.error "invalid user: #{current_user.errors.full_messages}"
+      return flash.alert = I18n.t('pages.languages.flash_messages.alert.locale_update_error')
+    end
+
+    unless @response.protocol_subscription.update has_language_input: true
+      Rails.logger.error "invalid protocol_subscription: #{@response.errors.full_messages}"
+      return flash.alert = I18n.t('pages.languages.flash_messages.alert.locale_update_error')
+    end
+
+    redirect_to change_params[:cb], notice: I18n.t('pages.languages.flash_messages.notice.locale_updated')
```

Comment:
can `change_params[:cb]` be nil? and if so, does `redirect_to nil` redirect to the root? Or does it not redirect? or does it give an error?

Reply:
not for my use case, but if this gets reused then potentially, yes. I'll think about what to do in that case, good catch

Reply:
it fails if it's nil. I've changed it to only redirect if a cb is given